instantsearch.py

```diff
@@ -102,13 +104,32 @@ def instantsearch(self):
         self.title_match_char = self.plugin.preferences['title_match_char']
         self.start_search_length = self.plugin.preferences['start_search_length']
         self.keystroke_delay = self.plugin.preferences['keystroke_delay']
+        self.open_when_unique = self.plugin.preferences['open_when_unique']
+        self.old_zim = self.plugin.preferences['old_zim']
 
         # building quick title cache
         def build(start = ""):
-            for s in self.window.ui.notebook.pages.list_pages(Path(start or ":")):
-                start2 = (start + ":" if start else "") + s.basename
-                self.cached_titles.append((start2, start2.lower()))
-                build(start2)
+            if self.old_zim is False:
+                for s in self.window.ui.notebook.pages.list_pages(Path(start or ":")):
```

Comment:
I think it would be better idea to determine the version change programatically. `old_zim` wouldn't be needed anymore.

```python
if pages in self.window.ui.notebook: # 
    o = self.window.ui.notebook.pages
else: # for zim0.66-
    o = self.window.ui.notebook.index
for s in o.list_pages(Path(start or ":")):
    ...
```